openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -409,9 +410,9 @@ public static OpenAPI getOpenAPIFromOpenAPIV3Parser(Path definitionPath) throws
     }
 
     /**
-     * Generate BallerinaMediaType for all the mediaTypes.
+     * Generate BallerinaMediaType for all the return mediaTypes.
      */
-    public static String getBallerinaMediaType(String mediaType) {
+    public static String getBallerinaMediaTypeForReturn(String mediaType) {
```

Comment:
Shall we implement one common util with return type `Optional<String>`, instead of repeating same content in these two functions `getBallerinaMediaTypeForReturn(String mediaType))`, `getBallerinaMediaTypeForRequest(String mediaType)` because the return and request have same media type support from ballerina when going to bind data.
 

- We can return an empty optional type if in case we encounter any unhandled scenario (most of the case default behaviour) then we can do an optional check where this util uses and do the rest of the string allocation according to the result.
ex: when util use in return type generator
`string returnType = getBallerinaMediaType(type).isPresent() ?  getBallerinaMediaType(type).get() : HTTP_RESPONSE `

Reply:
This method `getBallerinaMediaType` has 13 usages and in each place doing the check for `wildcard` media types seems redundant. Therefore added a flag to identify whether the mapping the for a request or for a response. Please review.

Reply:
+1